File: java/wsfl/src/main/java/wsfl/Timeit.java
package wsfl;

public class Timeit {

	private String name;
	private int number;
	
	public String getName() {
		return name;
	}

	public int getNumber() {
		return number;
	}

	public void set(String name, int number) {
		this.name = name;
		this.number = number;
	}
	
	
	public static void timeit(TimeitCallable f) throws Exception {
		Timeit timeit = new Timeit();
		long t = System.currentTimeMillis();		
		f.call(timeit);
		t = System.currentTimeMillis() - t;
		int n = timeit.number;
		double mean = n > 0 ? (double)t / n : 0;
		System.out.println(timeit.name + " " + t + " " + mean);
	}
}

File: python/wsfl/timeit.py
import time

def timeit(f):
    def _wrapper():
        t = time.time()
        n = f()
        t = (time.time() - t)*1000

        print('time', f.__name__, t, '(ms)', t/n if n>0 else None, '(ms)')
    return _wrapper